content/en/tutorials/working-with-grpc-client.md

```diff
@@ -103,20 +103,20 @@ If the user loses a device _(unregistered / no more shared with the user)_, the
 
 ### Get Resource from Device
 
-The `GetResourceFromDevice` retrieves resource content directly from the device - resource shadow value is not returned.
+The `GetResourceFromDevice` retrieves resource content directly from the device - resource shadow value is not returned. To define the expiration of command, set `GetResourceFromDeviceRequest.time_to_live` in nanoseconds(minimal 100ms). If the pending event is expired `GetResourceFromDevice.valid_until` (Unix timestamp in nanoseconds, 0 means forever), the cloud skips it and will be removed by creating a new snapshot event.
```

Comment:
100ms is not in nanoseconds.

Reply:
Why you mix two different informations together - if pending event is expired and in bracket you mention what 0 means?

It's not a pending event, it's pending command.

It will be removed when new snapshot creation occurs, which is configurable [here]...

Reply:
100ms is not in nanoseconds - is more readable than 100 000 000ns
It's not a pending event, it's pending command. - it is converted to an event. We don't have pending command.

Reply:
> 100ms is not in nanoseconds - is more readable than 100 000 000ns

Agree, but then document the possibilities properly. You wrote that time_to_live must be in nanoseconds and then you write 100ms. It's not understandable for readers who are not yet aware that time.Duration is used in the background.

>  it is converted to an event

This is API documentation. The user is not retrieving pending events, he is retrieving, using our API, pending commands.